### ISSUE TITLE
Issue with headerRF.txt files generated in case of multi-group Microsat/sequence data

### ISSUE THREAD

**Issue by gdurif** [diyabc/diyabc#43]
The `group summary statistics` section in generated `headerRF.txt` has an issue when dealing with multiple groups of Microsat of Sequence loci.

The `group summary statistics` titles is repeated for each groups, and the total number of summary statistics is each number of group summary statistics instead of the total across groups.

Here is an example:
```
group summary statistics (76)
group G1 (76)
NAL 1 2 3 4
HET 1 2 3 4
VAR 1 2 3 4
MGW 1 2 3 4
N2P 1.2 1.3 1.4 2.3 2.4 3.4
H2P 1.2 1.3 1.4 2.3 2.4 3.4
V2P 1.2 1.3 1.4 2.3 2.4 3.4
FST 1.2 1.3 1.4 2.3 2.4 3.4
LIK 1.2 2.1 1.3 3.1 1.4 4.1 2.3 3.2 2.4 4.2 3.4 4.3
DAS 1.2 1.3 1.4 2.3 2.4 3.4
DM2 1.2 1.3 1.4 2.3 2.4 3.4
AML 1.2.3 2.1.3 3.1.2 1.2.4 2.1.4 4.1.2 1.3.4 3.1.4 4.1.3 2.3.4 3.2.4 4.2.3

group summary statistics (76)
group G2 (76)
NAL 1 2 3 4
HET 1 2 3 4
VAR 1 2 3 4
MGW 1 2 3 4
N2P 1.2 1.3 1.4 2.3 2.4 3.4
H2P 1.2 1.3 1.4 2.3 2.4 3.4
V2P 1.2 1.3 1.4 2.3 2.4 3.4
FST 1.2 1.3 1.4 2.3 2.4 3.4
LIK 1.2 2.1 1.3 3.1 1.4 4.1 2.3 3.2 2.4 4.2 3.4 4.3
DAS 1.2 1.3 1.4 2.3 2.4 3.4
DM2 1.2 1.3 1.4 2.3 2.4 3.4
AML 1.2.3 2.1.3 3.1.2 1.2.4 2.1.4 4.1.2 1.3.4 3.1.4 4.1.3 2.3.4 3.2.4 4.2.3
```
and how it should be:
```
group summary statistics (152)
group G1 (76)
NAL 1 2 3 4
HET 1 2 3 4
VAR 1 2 3 4
MGW 1 2 3 4
N2P 1.2 1.3 1.4 2.3 2.4 3.4
H2P 1.2 1.3 1.4 2.3 2.4 3.4
V2P 1.2 1.3 1.4 2.3 2.4 3.4
FST 1.2 1.3 1.4 2.3 2.4 3.4
LIK 1.2 2.1 1.3 3.1 1.4 4.1 2.3 3.2 2.4 4.2 3.4 4.3
DAS 1.2 1.3 1.4 2.3 2.4 3.4
DM2 1.2 1.3 1.4 2.3 2.4 3.4
AML 1.2.3 2.1.3 3.1.2 1.2.4 2.1.4 4.1.2 1.3.4 3.1.4 4.1.3 2.3.4 3.2.4 4.2.3
group G2 (76)
NAL 1 2 3 4
HET 1 2 3 4
VAR 1 2 3 4
MGW 1 2 3 4
N2P 1.2 1.3 1.4 2.3 2.4 3.4
H2P 1.2 1.3 1.4 2.3 2.4 3.4
V2P 1.2 1.3 1.4 2.3 2.4 3.4
FST 1.2 1.3 1.4 2.3 2.4 3.4
LIK 1.2 2.1 1.3 3.1 1.4 4.1 2.3 3.2 2.4 4.2 3.4 4.3
DAS 1.2 1.3 1.4 2.3 2.4 3.4
DM2 1.2 1.3 1.4 2.3 2.4 3.4
AML 1.2.3 2.1.3 3.1.2 1.2.4 2.1.4 4.1.2 1.3.4 3.1.4 4.1.3 2.3.4 3.2.4 4.2.3
```

For information, here is the full `headerRF.txt` file in this example:
```
pseudo_observed_DATASET_toy_example1_microsatellites_one_pop_multiple_samples_over_time_001.mss
3 parameters and 152 summary statistics

2 scenarios: 6 5
scenario 1 [0.5] (3)
Npresent
0 sample 1
50 sample 1
200 sample 1
500 sample 1
tbn VarNe 1 Npast
scenario 2 [0.5] (1)
Npast
0 sample 1
50 sample 1
200 sample 1
500 sample 1

historical parameters priors (3,0)
Npast N UN[10,50000,0,0]
Npresent N UN[10,1000,0,0]
tbn T UN[10,1000,0,0]

loci description (50)
Locus_M_A_1_ <A> [M] G1 2 40
Locus_M_A_2_ <A> [M] G1 2 40
Locus_M_A_3_ <A> [M] G1 2 40
Locus_M_A_4_ <A> [M] G1 2 40
Locus_M_A_5_ <A> [M] G1 2 40
Locus_M_A_6_ <A> [M] G1 2 40
Locus_M_A_7_ <A> [M] G1 2 40
Locus_M_A_8_ <A> [M] G1 2 40
Locus_M_A_9_ <A> [M] G1 2 40
Locus_M_A_10_ <A> [M] G1 2 40
Locus_M_A_11_ <A> [M] G1 2 40
Locus_M_A_12_ <A> [M] G1 2 40
Locus_M_A_13_ <A> [M] G1 2 40
Locus_M_A_14_ <A> [M] G1 2 40
Locus_M_A_15_ <A> [M] G1 2 40
Locus_M_A_16_ <A> [M] G1 2 40
Locus_M_A_17_ <A> [M] G1 2 40
Locus_M_A_18_ <A> [M] G1 2 40
Locus_M_A_19_ <A> [M] G1 2 40
Locus_M_A_20_ <A> [M] G1 2 40
Locus_M_A_21_ <A> [M] G1 2 40
Locus_M_A_22_ <A> [M] G1 2 40
Locus_M_A_23_ <A> [M] G1 2 40
Locus_M_A_24_ <A> [M] G1 2 40
Locus_M_A_25_ <A> [M] G1 2 40
Locus_M_A_26_ <A> [M] G1 2 40
Locus_M_A_27_ <A> [M] G1 2 40
Locus_M_A_28_ <A> [M] G1 2 40
Locus_M_A_29_ <A> [M] G1 2 40
Locus_M_A_30_ <A> [M] G1 2 40
Locus_M_A_31_ <A> [M] G1 2 40
Locus_M_A_32_ <A> [M] G1 2 40
Locus_M_A_33_ <A> [M] G1 2 40
Locus_M_A_34_ <A> [M] G1 2 40
Locus_M_A_35_ <A> [M] G1 2 40
Locus_M_A_36_ <A> [M] G1 2 40
Locus_M_A_37_ <A> [M] G1 2 40
Locus_M_A_38_ <A> [M] G1 2 40
Locus_M_A_39_ <A> [M] G1 2 40
Locus_M_A_40_ <A> [M] G1 2 40
Locus_M_A_41_ <A> [M] G1 2 40
Locus_M_A_42_ <A> [M] G1 2 40
Locus_M_A_43_ <A> [M] G1 2 40
Locus_M_A_44_ <A> [M] G1 2 40
Locus_M_A_45_ <A> [M] G1 2 40
Locus_M_M_46_ <M> [M] G2 2 40
Locus_M_M_47_ <M> [M] G2 2 40
Locus_M_M_48_ <M> [M] G2 2 40
Locus_M_M_49_ <M> [M] G2 2 40
Locus_M_M_50_ <M> [M] G2 2 40

group priors (2)
group G1 [M]
MEANMU UN[1e-4,1e-3,5e-4,2]
GAMMU GA[1e-5,1e-2,Mean_u,2]
MEANP UN[1e-1,3e-1,2.2e-1,2]
GAMP GA[1e-2,9e-1,Mean_P,2]
MEANSNI UN[1e-8,1e-5,1e-7,2]
GAMSNI GA[1e-9,1e-4,Mean_u_SNI,2]
group G2 [M]
MEANMU UN[5e-4,5e-3,5e-4,2]
GAMMU GA[1e-5,1e-2,Mean_u,2]
MEANP UN[1e-1,3e-1,2.2e-1,2]
GAMP GA[1e-2,9e-1,Mean_P,2]
MEANSNI UN[1e-7,1e-4,1e-7,2]
GAMSNI GA[1e-9,1e-4,Mean_u_SNI,2]

group summary statistics (76)
group G1 (76)
NAL 1 2 3 4
HET 1 2 3 4
VAR 1 2 3 4
MGW 1 2 3 4
N2P 1.2 1.3 1.4 2.3 2.4 3.4
H2P 1.2 1.3 1.4 2.3 2.4 3.4
V2P 1.2 1.3 1.4 2.3 2.4 3.4
FST 1.2 1.3 1.4 2.3 2.4 3.4
LIK 1.2 2.1 1.3 3.1 1.4 4.1 2.3 3.2 2.4 4.2 3.4 4.3
DAS 1.2 1.3 1.4 2.3 2.4 3.4
DM2 1.2 1.3 1.4 2.3 2.4 3.4
AML 1.2.3 2.1.3 3.1.2 1.2.4 2.1.4 4.1.2 1.3.4 3.1.4 4.1.3 2.3.4 3.2.4 4.2.3

group summary statistics (76)
group G2 (76)
NAL 1 2 3 4
HET 1 2 3 4
VAR 1 2 3 4
MGW 1 2 3 4
N2P 1.2 1.3 1.4 2.3 2.4 3.4
H2P 1.2 1.3 1.4 2.3 2.4 3.4
V2P 1.2 1.3 1.4 2.3 2.4 3.4
FST 1.2 1.3 1.4 2.3 2.4 3.4
LIK 1.2 2.1 1.3 3.1 1.4 4.1 2.3 3.2 2.4 4.2 3.4 4.3
DAS 1.2 1.3 1.4 2.3 2.4 3.4
DM2 1.2 1.3 1.4 2.3 2.4 3.4
AML 1.2.3 2.1.3 3.1.2 1.2.4 2.1.4 4.1.2 1.3.4 3.1.4 4.1.3 2.3.4 3.2.4 4.2.3

scenario Npast Npresent tbn mumic_1 pmic_1 snimic_1 mumic_2 pmic_2 snimic_2   NAL_1_1       NAL_1_2       NAL_1_3       NAL_1_4       HET_1_1       HET_1_2       HET_1_3       HET_1_4       VAR_1_1       VAR_1_2       VAR_1_3       VAR_1_4       MGW_1_1       MGW_1_2       MGW_1_3       MGW_1_4      N2P_1_1.2     N2P_1_1.3     N2P_1_1.4     N2P_1_2.3     N2P_1_2.4     N2P_1_3.4     H2P_1_1.2     H2P_1_1.3     H2P_1_1.4     H2P_1_2.3     H2P_1_2.4     H2P_1_3.4     V2P_1_1.2     V2P_1_1.3     V2P_1_1.4     V2P_1_2.3     V2P_1_2.4     V2P_1_3.4     FST_1_1.2     FST_1_1.3     FST_1_1.4     FST_1_2.3     FST_1_2.4     FST_1_3.4     LIK_1_1.2     LIK_1_2.1     LIK_1_1.3     LIK_1_3.1     LIK_1_1.4     LIK_1_4.1     LIK_1_2.3     LIK_1_3.2     LIK_1_2.4     LIK_1_4.2     LIK_1_3.4     LIK_1_4.3     DAS_1_1.2     DAS_1_1.3     DAS_1_1.4     DAS_1_2.3     DAS_1_2.4     DAS_1_3.4     DM2_1_1.2     DM2_1_1.3     DM2_1_1.4     DM2_1_2.3     DM2_1_2.4     DM2_1_3.4    AML_1_1.2.3   AML_1_2.1.3   AML_1_3.1.2   AML_1_1.2.4   AML_1_2.1.4   AML_1_4.1.2   AML_1_1.3.4   AML_1_3.1.4   AML_1_4.1.3   AML_1_2.3.4   AML_1_3.2.4   AML_1_4.2.3     NAL_2_1       NAL_2_2       NAL_2_3       NAL_2_4       HET_2_1       HET_2_2       HET_2_3       HET_2_4       VAR_2_1       VAR_2_2       VAR_2_3       VAR_2_4       MGW_2_1       MGW_2_2       MGW_2_3       MGW_2_4      N2P_2_1.2     N2P_2_1.3     N2P_2_1.4     N2P_2_2.3     N2P_2_2.4     N2P_2_3.4     H2P_2_1.2     H2P_2_1.3     H2P_2_1.4     H2P_2_2.3     H2P_2_2.4     H2P_2_3.4     V2P_2_1.2     V2P_2_1.3     V2P_2_1.4     V2P_2_2.3     V2P_2_2.4     V2P_2_3.4     FST_2_1.2     FST_2_1.3     FST_2_1.4     FST_2_2.3     FST_2_2.4     FST_2_3.4     LIK_2_1.2     LIK_2_2.1     LIK_2_1.3     LIK_2_3.1     LIK_2_1.4     LIK_2_4.1     LIK_2_2.3     LIK_2_3.2     LIK_2_2.4     LIK_2_4.2     LIK_2_3.4     LIK_2_4.3     DAS_2_1.2     DAS_2_1.3     DAS_2_1.4     DAS_2_2.3     DAS_2_2.4     DAS_2_3.4     DM2_2_1.2     DM2_2_1.3     DM2_2_1.4     DM2_2_2.3     DM2_2_2.4     DM2_2_3.4    AML_2_1.2.3   AML_2_2.1.3   AML_2_3.1.2   AML_2_1.2.4   AML_2_2.1.4   AML_2_4.1.2   AML_2_1.3.4   AML_2_3.1.4   AML_2_4.1.3   AML_2_2.3.4   AML_2_3.2.4   AML_2_4.2.3  

```

**Comment by gdurif** [comment]
Note: the multi-group case can also be encountered with SNP IndSeq data. We are checking if this issue is also encountered in this case.

**Comment by fradav** [comment]
This is historical code from JMC and as it is closely coupled to his own (very fragile) parsing code of the headers, I strongly advise to let as it is. Unless it is an issue for diyabcGUI ?

**Comment by gdurif** [comment]
It is an issue for `diyabcGUI` because I cannot use a user-provided headerRF.txt file to increase the reftable since `diyabc` returns a segfault with such header.

If it is too complicated, I can design a work-around to regenerate the `header.txt` file for `diyabc` instead of copying the generated `headerRF.txt` file.

**Comment by fradav** [comment]
Ok, can you provide me a full example with two locii groups ? (header + dataset) so I could add it to the test list ?

**Comment by fradav** [comment]
Fixed in 6f79f6e
Release : https://github.com/diyabc/diyabc/releases/tag/v1.1.17
